strawberryfields/ops.py

```diff
@@ -587,8 +587,11 @@ class Coherent(Preparation):
         :math:`\mathbf{V}= \frac{\hbar}{2} I`.
     """
 
-    def __init__(self, r=0.0, p=0.0):
-        super().__init__([r, p])
+    def __init__(self, r=0.0, phi=0.0):
+        if (np.iscomplex([r, phi])).any():
+            raise ValueError("The arguments of Coherent(r, phi) cannot be complex")
+
+        super().__init__([r, phi])
```

Comment:
what is this `super()` ?

Reply:
`super()` creates an instance of the parent class, `Preparation`. With this syntax we can place a call to `Preparation.__init__`, running the `__init__` method of the parent class.

This was something originally there too, the addition would be the error being raised.